### ISSUE TITLE
GetName() Still broken and idea behind why.

### ISSUE THREAD

**Issue by TaylorSasser** [KN4CK3R/UnrealEngineSDKGenerator#5]
Hey, i talked to domo and he is having the same issue with the GetName methods. For some reason with your SDK when iterating through all the GNames the game will print nothing or crash when the FName constructor is called and it starts to iterate through the GNames. I suspect that (See reference one of the hastebin) that the FNameEntry class is somehow passing the nullptr check but in reality is null. I am not sure how it does, but when your second for loop starts to increment through the GNames searching for the appropriate name to find, it always will get the QWordProperty's FNameEntry without any problem. Directly after the QWordProperty is where the trouble starts, because according to the NameDump there are 4 iterations of the for loop before the next FNameEntry is found, which is at spot 20. 

tldr; nullptr check is not working properly, and the GNames array has many null or bad FNames inside of it

[NamesDump.txt](https://github.com/KN4CK3R/UnrealEngineSDKGenerator/files/781815/NamesDump.txt)
https://hastebin.com/bayeyudane.m - See this for the suspected problem
https://hastebin.com/opubukoxut.cpp - Should illustrate the problem

**Comment by KN4CK3R** [comment]
Which game?

**Comment by KN4CK3R** [comment]
I don't get the problem. The name array could contain nullptrs so you need to check for them. In your example:
```
void TestGNamesCrashesGame()
{
	for (int i = 0; i < 50; i++)
	{
		auto temp = FName::GetGlobalNames()[i];
		if (temp != nullptr)
		{
			std::string Name = temp->GetName();
			std::cout << Name << std::endl;
		}
	}
}
```
It that your problem? What code leads to a crash? Your code with the SDK or code from the generator?

**Comment by TaylorSasser** [comment]
This is for rocket league, and the1domo says it also crashes with unreal tournament 3. What leads to the crash is some error inside the GNames containing null ptr. It is really hard to tell what the problem but all I can say that somewhere along the lines of calling the UFunctions GetName method it breaks and the game crashes.

**Comment by KN4CK3R** [comment]
I added a repository with a testproject.
https://github.com/KN4CK3R/UnrealEngineSDKGenerator-Test
Uses just the generated SDK and a ProcessEvent hook which prints the caller names. Can't reproduce an error.

**Comment by TaylorSasser** [comment]
Still having a problem, and no idea why. GetName still crashes and it legit drives me insane.

**Comment by KN4CK3R** [comment]
Then give me code to reproduce. I can't help you if you just say you got crashes. Attach a debugger after the crash and look at the stack dump, etc...

**Comment by TaylorSasser** [comment]
Here is a video where i explain whats going on. it should help. I changed my PE hooking method to yours and that fixed all the crashes but the UFunction doesnt pass the nullptr check. Apologies for my bad commentary.
https://youtu.be/kj8MiQDIeT8

**Comment by KN4CK3R** [comment]
If I have understood correctly you remove the vtable pointer from the UObject class? If so don't remove the pointer, it's not needed for my generator. If you remove the vtable pointer you change the class layout so the name object offset is wrong and GetName may crash.

**Comment by TaylorSasser** [comment]
The other method I used previously requires you too remove the vtable pointer but i have had it enabled since i switched to your method of hooking.

**Comment by KN4CK3R** [comment]
Then I don't know how to help you. I don't have Rocket League so I can't test it for myself. If you could family share it on STEAM I can test it next weekend.

**Comment by TaylorSasser** [comment]
Add me on discord, I have several accounts. Taylor#8660

**Comment by KN4CK3R** [comment]
I don't have discord, I will write you a pm at uc.
